src/main/java/carpet/commands/PlayerCommand.java

```diff
@@ -365,6 +366,11 @@ private static int shadow(CommandContext<CommandSourceStack> context)
             Messenger.m(context.getSource(), "r Cannot shadow fake players");
             return 0;
         }
+        MinecraftServer server = Objects.requireNonNull(player.getServer());
```

Comment:
Don't think the server can ever be null for a ServerPlayer.

Reply:
Yeah, I've even seen this in action plenty already I think. Too trusting of what IntelliJ is telling me, time to disable some inspections.